libp2p/multistream.nim

```diff
@@ -98,19 +101,29 @@ proc select*(_: MultistreamSelect | type MultistreamSelect,
       # No alternatives, fail
       return ""
 
-proc select*(_: MultistreamSelect | type MultistreamSelect,
-             conn: Connection,
-             proto: string): Future[bool] {.async.} =
+proc select*(
+    _: MultistreamSelect | type MultistreamSelect,
+    conn: Connection,
+    proto: string
+): Future[bool] {.async: (raises: [
+    CancelledError, LPStreamError, MultiStreamError]).} =
```

Comment:
not sure this is the right PR for it, but reraising underlying stream errors represents an abstraction leak - ie they should be caught and rebranded as multistreamerror _really_

Reply:
Agree, but the pattern of re-raising `LPStreamError` applies to quite large parts of the codebase, and the goal for now is to just document the reality with the `{.async: (raises).}`, while fixing the obvious offenders of `raise (ref CatchableError)()`. Semantic changes should be separate.

Reply:
Another precedent we have in `muxer.nim` where the procs currently raise `{.async: (raises: [CancelledError, LPStreamError, MuxerError].}`, it's just how it's done today.